### PR TITLE
Set alabaster theme default config right in conf.py template (fix #3987)

### LINKED ISSUE #3987: Changing theme from alabaster causes HTML build to fail
Subject: Building HTML fails if theme is not alabaster

### Problem
- Can't use any theme that is not alabaster

#### Procedure to reproduce the problem
```
- Change theme to something other than alabaster
```

#### Error logs / results
```
# Sphinx version: 1.6.3
# Python version: 3.6.1 (CPython)
# Docutils version: 0.14 
# Jinja2 version: 2.9.6
# Last messages:
#   checking consistency...
#   
#   done
#   
#   preparing documents...
#   
#   done
#   
#   writing output... [  5%] about
#   
# Loaded extensions:
#   alabaster (0.7.10) from C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\alabaster\__init__.py
#   sphinx.ext.autodoc (1.6.3) from C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\ext\autodoc.py
#   sphinx.ext.doctest (1.6.3) from C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\ext\doctest.py
#   sphinx.ext.todo (1.6.3) from C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\ext\todo.py
#   sphinx.ext.mathjax (1.6.3) from C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\ext\mathjax.py
#   sphinx.ext.viewcode (1.6.3) from C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\ext\viewcode.py
Traceback (most recent call last):
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\cmdline.py", line 306, in main
    app.build(opts.force_all, filenames)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\application.py", line 339, in build
    self.builder.build_update()
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\builders\__init__.py", line 331, in build_update
    'out of date' % len(to_build))
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\builders\__init__.py", line 397, in build
    self.write(docnames, list(updated_docnames), method)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\builders\__init__.py", line 434, in write
    self._write_serial(sorted(docnames))
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\builders\__init__.py", line 443, in _write_serial
    self.write_doc(docname, doctree)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\builders\html.py", line 561, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\builders\html.py", line 945, in handle_page
    output = self.templates.render(templatename, ctx)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\jinja2glue.py", line 176, in render
    return self.environment.get_template(template).render(context)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\jinja2\asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\jinja2\environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\themes\basic\page.html", line 10, in top-level template code
    {%- extends "layout.html" %}
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\themes\basic\layout.html", line 175, in top-level template code
    {%- block content %}
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\themes\basic\layout.html", line 193, in block "content"
    {%- block sidebar2 %}{{ sidebar() }}{% endblock %}
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\themes\basic\layout.html", line 193, in block "sidebar2"
    {%- block sidebar2 %}{{ sidebar() }}{% endblock %}
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\jinja2\sandbox.py", line 427, in call
    return __context.call(__obj, *args, **kwargs)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\jinja2\runtime.py", line 553, in _invoke
    rv = self._func(*arguments)
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\themes\basic\layout.html", line 64, in template
    {%- include sidebartemplate %}
  File "C:\Users\591709\Documents\Projects\icp-data-tool\lib\site-packages\sphinx\jinja2glue.py", line 200, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: about.html
```

#### Expected results
- Build html

### Reproducible project / your project
- <link to your project, or attach zipped small project sample>

### Environment info
- OS: <Unix/Linux/Mac/Win/other with version> Win 10
- Python version: 3.6.1
- Sphinx version: 1.6.3
- <Extra tools e.g.: Browser, tex or something else>

### PR DESCRIPTION
**Pull request by jfbu**
### Relates
- #4002, #3987

### CODE REVIEW

File: sphinx/templates/quickstart/conf.py_t [hunk 1]
```diff
@@ -110,12 +114,15 @@ html_static_path = ['{{ dot }}static']
...
-# This is required for the alabaster theme
-# refs: http://alabaster.readthedocs.io/en/latest/installation.html#sidebars
+# For alabaster, check
+# http://alabaster.readthedocs.io/en/latest/installation.html#sidebars
 html_sidebars = {
     '**': [
-        'relations.html',  # needs 'show_related': True theme option to display
-        'searchbox.html',
+        'about.html',          # comment this out if not using alabaster theme
+        #  'navigation.html',  # specific to alabaster (and possibly others)
```
Comment (jfbu) [inline sphinx/templates/quickstart/conf.py_t]: perhaps this one should be uncommented-out too with instructions to comment it out if not using alabaster theme

**Comment by tk0miya** [comment]
I think this brings confusion similar to #3987 for users again.
I prefer to add `default_sidebars` setting to `theme.conf`. It allows HTML themes to specify its own custom sidebars through configurations.
What do you think?

**Comment by jfbu** [comment]
@tk0miya I agree pushing out `html_sidebars` default config (and `html_theme_options`) to theme configuration is way to go. We must provide correct `html_sidebars` setting for `alabaster` if the latter is our default theme, but user should be able to switch theme easily, only by setting `html_theme` and not have to worry about theme config, which should be the "right one" without further ado and each theme should provide its own correct minimal default config not needing user intervention.

**Comment by tk0miya** [comment]
Now I proposed #4177 instead.

**Pull request closed without merging**